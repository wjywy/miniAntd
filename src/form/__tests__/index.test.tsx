import * as React from 'react';

import { render } from '@testing-library/react';
import Form from '../';
import Input from '../../input';
import mountTest from '../../tests/mount';

describe('Form', () => {
  mountTest(Form);

  it('no Style on Form', () => {
    // 应该form和item都没样式
    const onChange = jest.fn();

    const { container } = render(
      <Form noStyle={true}>
        <Form.Item
          name="test"
          initialValue="bamboo"
          rules={[{ required: true }]}
        >
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(container.querySelectorAll('.ci-form-item').length).toBe(0);
    expect(container.querySelectorAll('.ci-form').length).toBe(0);
  });

  it('no Style on Form.Item', () => {
    // 应该form和item都没样式
    const onChange = jest.fn();

    const { container } = render(
      <Form>
        <Form.Item
          name="test"
          noStyle={true}
          initialValue="bamboo"
          rules={[{ required: true }]}
        >
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(container.querySelectorAll('.ci-form-item').length).toBe(0);
    expect(container.querySelectorAll('.ci-form').length > 0).toBeTruthy();
  });

  it('show require icon when required is true or rules contain required', () => {
    const onChange = jest.fn();
    const { container, rerender } = render(
      <Form>
        <Form.Item
          name="test"
          initialValue="bamboo"
          rules={[{ required: true }]}
        >
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(
      container.querySelectorAll('.ci-form-item-label-required').length > 0,
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
    rerender(
      <Form>
        <Form.Item name="test" initialValue="bamboo" required={true}>
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(
      container.querySelectorAll('.ci-form-item-label-required').length > 0,
    ).toBeTruthy();
  });

  it('show different size', () => {
    const onChange = jest.fn();
    const { container, rerender } = render(
      <Form size="small">
        <Form.Item
          name="test"
          initialValue="bamboo"
          rules={[{ required: true }]}
        >
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(container.firstChild).toMatchSnapshot();

    rerender(
      <Form size="large">
        <Form.Item
          name="test"
          initialValue="bamboo"
          rules={[{ required: true }]}
        >
          <Input onChange={onChange} />
        </Form.Item>
      </Form>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
