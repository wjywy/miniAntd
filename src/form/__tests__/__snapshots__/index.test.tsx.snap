// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form show different size 1`] = `
<form
  class="ci-form ci-form-horizontal ci-form-small"
>
  <div
    value="bamboo"
  >
    <div
      class="ci-form-item"
    >
      <label
        class="ci-form-item-label ci-form-item-label-required"
        title="test"
      >
        test
        :
      </label>
      <div
        class="ci-form-item-content"
      >
        <input
          class="input_basic small normal"
          type="text"
        />
      </div>
    </div>
    <div
      class="ci-error-list"
      role="alert"
      style="opacity: 0;"
    />
  </div>
</form>
`;

exports[`Form show different size 2`] = `
<form
  class="ci-form ci-form-horizontal ci-form-large"
>
  <div
    value="bamboo"
  >
    <div
      class="ci-form-item"
    >
      <label
        class="ci-form-item-label ci-form-item-label-required"
        title="test"
      >
        test
        :
      </label>
      <div
        class="ci-form-item-content"
      >
        <input
          class="input_basic large normal"
          type="text"
        />
      </div>
    </div>
    <div
      class="ci-error-list"
      role="alert"
      style="opacity: 0;"
    />
  </div>
</form>
`;

exports[`Form show require icon when required is true or rules contain required 1`] = `
<form
  class="ci-form ci-form-horizontal ci-form-small"
>
  <div
    value="bamboo"
  >
    <div
      class="ci-form-item"
    >
      <label
        class="ci-form-item-label ci-form-item-label-required"
        title="test"
      >
        test
        :
      </label>
      <div
        class="ci-form-item-content"
      >
        <input
          class="input_basic small normal"
          type="text"
        />
      </div>
    </div>
    <div
      class="ci-error-list"
      role="alert"
      style="opacity: 0;"
    />
  </div>
</form>
`;
